src/erlcloud_sns.erl

```diff
@@ -624,8 +624,8 @@ parse_key("EventEndpointUpdated") -> event_endpoint_updated;
 parse_key("EVentDeliveryFailure") -> event_delivery_failure;
 parse_key(OtherKey) -> list_to_atom(string:to_lower(OtherKey)).
 
-scheme_to_protocol(S) when is_list(S) -> s2p(string:to_lower(S));
-scheme_to_protocol(_)                 -> erlang:error({sns_error, badarg}).
+-spec scheme_to_protocol(string()) -> string().
+scheme_to_protocol(S) when is_list(S) -> s2p(string:to_lower(S)).
```

Comment:
this change breaks the tests. 
I know that sns/s3 API with use `erlang:error` is awful but that is subject to further major revision change and not to be addressed within this PR